tests/Tests/x__pipelines/x__data-science-pipelines-api.robot

```diff
@@ -0,0 +1,46 @@
+*** Settings ***
+Documentation       Test suite for OpenShift Pipeline
+Resource            ../../Resources/RHOSi.resource
+Resource            ../../Resources/ODS.robot
+Resource            ../../Resources/Common.robot
+Library             DateTime
+Library             ../../../libs/OpenShiftPipelineAPI.py
+Suite Setup         RHOSi Setup
+Suite Teardown      RHOSi Teardown
+
+
+*** Variables ***
+${KUBEFLOW_OPERATOR_YAML}=                     tests/Resources/Files/kubeflow-operator.yaml
+${DATA_SCIENCE_PIPELINES_APPLICATION_YAML}=    tests/Resources/Files/data-science-pipelines.yaml
+
+*** Test Cases ***
+Install Kubeflow Operator
+    [Documentation]    Install and verifies that Kubeflow Operator is installed and working
+    [Tags]    Smoke
+    Oc Apply    kind=KfDef    src=${KUBEFLOW_OPERATOR_YAML}     namespace=redhat-ods-applications
+    ${pod_count}    Run    oc get pods -n redhat-ods-applications | grep data-science-pipelines-operator-controller-manager | wc -l
+    Should Be True    ${pod_count} == 3    All the pods were created
+
+
+Install DataSciencePipelinesApplication CRD
+    [Documentation]    Install and verifies that DataSciencePipelinesApplication CRD is installed and working
+    [Tags]    Smoke
+    Run    oc new-project ods-ci-pipeline
+    Oc Apply    kind=DataSciencePipelinesApplication    src=${DATA_SCIENCE_PIPELINES_APPLICATION_YAML}     namespace=ods-ci-pipeline
+    ${generation_value}    Run    oc get datasciencepipelinesapplications -o json | jq '.items[0].metadata.generation'
+    Should Be True    ${generation_value} == 1    DataSciencePipelinesApplication created
+
+
+Test and end-to-end pipeline workflow via API
+    [Documentation]    Create, run and double check the pipeline result. In the end, clean the pipeline resources.
+    [Tags]    Smoke
```

Comment:
This test should be in Tier1 and Sanity  (will be in Smoke once is stable). Missing Polarion ID. 

Also, it could be refactored with keywords to simplify the first level of abstraction (see https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit#heading=h.7gognio1srsr and we can discuss in a meeting too)